packages/io/io-utils/BinaryReader.js

```diff
@@ -1,51 +1,58 @@
 // BinaryReader
+// Converted to ES5 Class by @z3dev
 // Refactored by Vjeux <vjeuxx@gmail.com>
 // http://blog.vjeux.com/2010/javascript/javascript-binary-reader.html
 
 // Original
 // + Jonas Raoni Soares Silva
 // @ http://jsfromhell.com/classes/binary-deserializer [rev. #1]
 
-function BinaryReader (data) {
-  this._buffer = data
-  this._pos = 0
-}
+class BinaryReader {
+  /*
+   * Construct a BinaryReader from the given data.
+   * The data is a string created from the specified sequence of UTF-16 code units.
+   * See String.fromCharCode()
+   * See _readByte() below
+   */
+  constructor (data) {
+    this._buffer = data
+    this._pos = 0
+  }
 
-BinaryReader.prototype = {
   /* Public */
-  readInt8: function () { return this._decodeInt(8, true) },
-  readUInt8: function () { return this._decodeInt(8, false) },
-  readInt16: function () { return this._decodeInt(16, true) },
-  readUInt16: function () { return this._decodeInt(16, false) },
-  readInt32: function () { return this._decodeInt(32, true) },
-  readUInt32: function () { return this._decodeInt(32, false) },
-
-  readFloat: function () { return this._decodeFloat(23, 8) },
-  readDouble: function () { return this._decodeFloat(52, 11) },
-
-  readChar: function () { return this.readString(1) },
-  readString: function (length) {
+  readInt8 () { return this._decodeInt(8, true) }
```

Comment:
Should these be arrow functions to be consistent with the rest of JSCAD?

Reply:
Will give it a try but these are ‘methods’ on a ‘class’, which is different then arrow functions which have no ‘this’ context.

It’s a strange language, JavaScript.

Reply:
Damn, yeah, you're right.  *shakes fist* Javascript!